### PR TITLE
move common prompt to external function

### PR DESCRIPTION
**Pull request by Eriner**

### CODE REVIEW

File: modules/prompt/functions/promptpwd [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#
+# prompt setup function common to many prompts
+# moved to external function to ruduce rudundant code
+#
+local prompt="$1"
+local pwd="${PWD/#$HOME/~}"
+ 
+if [[ "$pwd" == (#m)[/~] ]]; then
+	# because this function is sourced from other scripts, the vars must be exported
+  export _prompt_"${prompt}"_pwd="$MATCH"
```
Comment (sorin-ionescu) [inline modules/prompt/functions/promptpwd]: Why are you exporting it?

**Comment by Eriner** [comment]
I believe the comment is self-explanatory. To go into more detail, if it isn't exported, the variable isn't available to the calling function ex: `prompt_damoekri_setup`.

An alternative way of handling it would be to create `local dir` and assign what is currently being exported to that, then at the end of the function, `echo ${dir}`. Inside the prompt setups, they could just assign the output of that to a variable and use that instead.

If you know of a better way to handle it, please let me know and I will commit a change.

**Comment by Eriner** [comment]
This would close issue https://github.com/sorin-ionescu/prezto/issues/701

**Comment by amerlyq** [comment]
Hm, I have slightly different approach to this problem as whole -- I have splitted all prompt into keyparts to make it modular. See at [prompt](https://github.com/amerlyq/airy/blob/master/cfg/shell/zsh/prompt) and [prompt.d](https://github.com/amerlyq/airy/tree/master/cfg/shell/zsh/prompt.d). Not ideal, of course, but allows much more control.

**Comment by sorin-ionescu** [comment]
Your prompts will share styling. In this case, it's just one function, no styling.

File: modules/prompt/functions/promptpwd [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#
+# prompt setup function common to many prompts
+# moved to external function to ruduce code rudundancy
```
Comment (belak) [inline modules/prompt/functions/promptpwd]: Nit: should be "reduce"

**Comment by belak** [comment]
I'm merging this into a temp branch so I can make a tweak and cherry-pick it to master. I guess weird things happen when a repo is deleted while there are still PRs open.

**Comment by belak** [comment]
This has been merged into master along with some spelling fixes and a fix to ensure extendedglob is set for this function. Thanks for your contribution!

**Comment by sorin-ionescu** [comment]
Be careful with those functions. They are not always completely identical. I was trying to think of a naming scheme since promptpwd is not descriptive enough.

**Comment by belak** [comment]
Maybe shortpwd or something similar?

**Comment by sorin-ionescu** [comment]
It's still doesn't say how short it is and what's included.

**Pull request merged**